scalding-core/src/main/scala/com/twitter/scalding/ExecutionContext.scala

```diff
@@ -31,6 +36,25 @@ trait ExecutionContext {
   def flowDef: FlowDef
   def mode: Mode
 
+  def getIdentifierOpt(descriptions: Seq[String]): Option[String] = {
+    if (descriptions.nonEmpty) {
+      Some(descriptions.distinct.mkString(", "))
+    } else {
+      None
+    }
+  }
```

Comment:
nit: our style is to only use `{ }` when needed (more than one expression). Many of your `{ }` are redundant and we'd prefer if you not add them.